pkg/kubelet/dockertools/exec.go

```diff
@@ -90,15 +88,22 @@ func (*NsenterExecHandler) ExecInContainer(client DockerInterface, container *do
 		if stderr != nil {
 			command.Stderr = stderr
 		}
-
-		return command.Run()
+		if err := command.Start(); err != nil {
+			return err
+		}
+	}
+	if timeout > 0 {
```

Comment:
A timeout of zero means no timeout, same interpretation as in `http.Client` in the stdlib.